### PR TITLE
cms124v7

### PR DESCRIPTION
**Pull request by ducvtran**
cervical cancer screening

### CODE REVIEW

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+define "Numerator":
+	exists "Pap Test Within 3 Years"
+		or exists "Pap Test With HPV Within 5 Years"
+
+define "Denominator":
+	"Initial Population"
+
+define "Pap Test with Results":
+	[Observation: "Pap Test"] PapTest
+		where PapTest.value is not null
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: Add `and PapTest.status in { 'final', 'amended', 'corrected' }`

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+define "Pap Test Within 3 Years":
+	"Pap Test with Results" PapTest
+		where PapTest.relevantPeriod 3 years or less before end of "Measurement Period"
+
+define "Qualifying Encounters":
+	( [Encounter: "Office Visit"]
+		union [Encounter: "Preventive Care Services - Established Office Visit, 18 and Up"]
+		union [Encounter: "Preventive Care Services-Initial Office Visit, 18 and Up"]
+		union [Encounter: "Home Healthcare Services"] ) ValidEncounter
+		where ValidEncounter.period during "Measurement Period"
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: Add `and ValidEncounter.status = 'finished'`

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	"Initial Population"
+
+define "Pap Test with Results":
+	[Observation: "Pap Test"] PapTest
+		where PapTest.value is not null
+
+// DT need to investigate more
+define "Pap Test Within 3 Years":
+	"Pap Test with Results" PapTest
+		where PapTest.relevantPeriod 3 years or less before end of "Measurement Period"
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: relevantPeriod is a QDM element. Use FHIR Observation.effective element as discussed in mapping tables.

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	( [Encounter: "Office Visit"]
+		union [Encounter: "Preventive Care Services - Established Office Visit, 18 and Up"]
+		union [Encounter: "Preventive Care Services-Initial Office Visit, 18 and Up"]
+		union [Encounter: "Home Healthcare Services"] ) ValidEncounter
+		where ValidEncounter.period during "Measurement Period"
+
+define "PapTest Within 5 Years":
+	( "Pap Test with Results" PapTestOver30YearsOld
+			with ["Patient Characteristic Birthdate"] BirthDate
+				such that Global."CalendarAgeInYearsAt"(BirthDate.birthDatetime, start of PapTestOver30YearsOld.relevantPeriod)>= 30
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: Use the same method as in the Initial Population expression (Patient.birthDate). Make a TODO to use the birthTime extension instead (we'll address that later). Also, relevantPeriod is a QDM element. Use FHIR Observation.effective element as discussed in mapping tables.

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+					and PapTestOver30YearsOld.relevantPeriod 5 years or less before end of "Measurement Period"
+	)
+
+define "Pap Test With HPV Within 5 Years":
+	"PapTest Within 5 Years" PapTestOver30YearsOld
+		with [Observation: "HPV Test"] HPVTest
+			such that HPVTest.value is not null
+				and ( HPVTest.effective starts 1 day or less before or on day of start of PapTestOver30YearsOld.relevantPeriod
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: Add `and HPVTest in { 'final', 'amended', 'corrected' }`. Also, relevantPeriod is a QDM element. Use FHIR Observation.effective element as discussed in mapping tables.

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+				)
+
+define "Denominator Exclusions":
+	Hospice."Has Hospice"
+		or exists "Surgical Absence of Cervix"
+		or exists "Absence of Cervix"
+
+define "Absence of Cervix":
+	[Condition : "Congenital absence of cervix (disorder)"] NoCervixBirth
+		where NoCervixBirth.onset starts before end "Measurement Period"
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: Add `and NoCervixBirth.clinicalStatus = 'active'`

File: pages/cql/cms124_FHIR-7.2.000.cql [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+define "Surgical Absence of Cervix":
+	[Procedure: "Hysterectomy with No Residual Cervix"] NoCervixHysterectomy
+		where NoCervixHysterectomy.performed ends before end "Measurement Period"
```
Comment (c-schuler) [inline pages/cql/cms124_FHIR-7.2.000.cql]: Add `and NoCervixHistory.status = 'completed'`

File: resources/measure/measure-cms124.json [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+            }
+          ]
+        }
+      ],
+  "supplementalData": [
+    {
+      "identifier": {
+        "value": "supplemental-data-race"
+      },
+      "criteria": "CMS124.\"Supplemental Data - Race\""
```
Comment (c-schuler) [inline resources/measure/measure-cms124.json]: These expression references should be the same as what's in the CQL (e.g. CMS124.\"SDE Race\")

**Comment by ducvtran** [comment]
done

**Comment by c-schuler** [comment]
Great work, thank you!

**Pull request merged**